src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/DependenciesProjectTreeProvider.cs

```diff
@@ -0,0 +1,973 @@
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+using Microsoft.Build.Framework.XamlTypes;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.ProjectSystem.References;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies
+{
+    /// <summary>
+    /// Provides the special "Dependencies" folder to project trees.
+    /// </summary>
+    [Export(ExportContractNames.ProjectTreeProviders.PhysicalViewRootGraft, typeof(IProjectTreeProvider))]
+    [Export(typeof(IDependenciesGraphProjectContext))]
+    [AppliesTo(ProjectCapability.DependenciesTree)]
+    internal class DependenciesProjectTreeProvider : ProjectTreeProviderBase, 
+                                                     IProjectTreeProvider, 
+                                                     IDependenciesGraphProjectContext
+    {
+        private static readonly ProjectTreeFlags DependenciesRootNodeFlags
+                = ProjectTreeFlags.Create(ProjectTreeFlags.Common.BubbleUp
+                                          | ProjectTreeFlags.Common.ReferencesFolder
+                                          | ProjectTreeFlags.Common.VirtualFolder)
+                                  .Add("DependenciesRootNode");
+
+        // TODO: Consider moving this to an [AppliesTo] attribute and
+        // removing the GetIsApplicableAsync method or always returning true from it.
+        /// <summary>
+        /// The AppliesTo expression that must be satisfied for this subtree provider to be applicable.
+        /// </summary>
+        private static readonly string ApplicabilityTest 
+                = string.Format(CultureInfo.InvariantCulture, "{0} & ({1} | {2} | {3} | {4} | {5})",
+                                ProjectCapabilities.ReferencesFolder,
+                                ProjectCapabilities.AssemblyReferences,
+                                ProjectCapabilities.ComReferences,
+                                ProjectCapabilities.ProjectReferences,
+                                ProjectCapabilities.SdkReferences,
+                                ProjectCapabilities.WinRTReferences);
+
+        /// <summary>
+        /// Keeps latest updated snapshot of all rules schema catalogs
+        /// </summary>
+        private IImmutableDictionary<string, IPropertyPagesCatalog> NamedCatalogs { get; set; }
+
+        /// <summary>
+        /// Gets the collection of <see cref="IProjectTreePropertiesProvider"/> imports 
+        /// that apply to the references tree.
+        /// </summary>
+        [ImportMany(ReferencesProjectTreeCustomizablePropertyValues.ContractName)]
+        private OrderPrecedenceImportCollection<IProjectTreePropertiesProvider> ProjectTreePropertiesProviders { get; set; }
```

Comment:
THose ImportMany are to fill collection , will keep them out of constructor

---
In reply to: [74316851](https://github.com/dotnet/roslyn-project-system/pull/349#discussion_r74316851) [](ancestors = 74316851)